transport/transporters/rabbitmq/factory.go

```diff
@@ -0,0 +1,200 @@
+	channel, err := conn.Channel()
+	if err != nil {
+		log.WithError(err).Fatal("Could not open channel")
+	}
+	defer channel.Close()
+	err = channel.ExchangeDeclare(
+		conf.exchange, // name of the exchange
+		"topic",       // type
+		wabbit.Option{
+			"durable":  true,
+			"delete":   false,
+			"internal": false,
+			"noWait":   false,
+		},
+	)
+	log.WithField("exchangeName", conf.exchange).WithError(err).Fatal("Could not declare exchange")
+
+	transports := make([]*transport.Transporter, workers)
+
+	// Make and link transporters to the batcher's output channels
+	for i := 0; i < workers; i++ {
+		t := transporter.NewTransporter(shutdownHandler, inputChans[i], txnsWritten, statsChan, *log, i, conf.exchange, conn, conf.batchSize)
+		transports[i] = &t
+	}
+
+	go connectionCleanup(shutdownHandler, conn, log)
+
+	return transports
+}
+
+func getConfig(transportConfig map[string]interface{}) *config {
+	exchangeNameVar := transportConfig[ConfVarExchangeName]
+	exchangeName, ok := exchangeNameVar.(string)
+
+	if !ok {
+		log.Fatalf("Expected type for %s is %s", ConfVarExchangeName, "string")
+	}
+
+	usernameVar := transportConfig[ConfVarUsername]
+	var username string
+	if usernameVar != "" {
+		username, ok = usernameVar.(string)
+
+		if !ok {
+			log.Fatalf("Expected type for %s is %s", ConfVarUsername, "string")
+		}
+	}
+
+	passwordVar := transportConfig[ConfVarPassword]
+	var password string
+	if passwordVar != "" {
+		password, ok = passwordVar.(string)
+
+		if !ok {
+			log.Fatalf("Expected type for %s is %s", ConfVarPassword, "string")
+		}
+	}
+
+	hostVar := transportConfig[ConfVarHost]
+	var host string
+	if hostVar != "" {
+		host, ok = hostVar.(string)
+
+		if !ok {
+			log.Fatalf("Expected type for %s is %s", ConfVarHost, "string")
+		}
+	}
+
+	virtualHostVar := transportConfig[ConfVarVirtualHost]
+	var virtualHost string
+	if virtualHostVar != "" {
+		virtualHost, ok = virtualHostVar.(string)
+
+		if !ok {
+			log.Fatalf("Expected type for %s is %s", ConfVarVirtualHost, "string")
+		}
+	}
+
+	batchSizeVar := transportConfig[ConfVarWriteBatchSize]
+	batchSize, ok := batchSizeVar.(int)
+
+	if !ok {
+		log.Fatalf("Expected type for %s is %s", ConfVarWriteBatchSize, "int")
+	}
+
+	return &config{
+		host:      host,
+		username:  username,
+		password:  password,
+		vhost:     virtualHost,
+		exchange:  exchangeName,
+		batchSize: batchSize,
+	}
+}
+
+// createConnection either returns a connection or logs a fatal message
+func createConnection(host, user, pass, vhost string, log *logrus.Entry) wabbit.Conn {
+	amqpURL := &url.URL{
+		Scheme: "amqp",
+		Host:   host,
+		User:   url.UserPassword(user, pass),
+		Path:   vhost,
+	}
+	conn, err := amqp.Dial(amqpURL.String())
+	if err != nil {
+		log.WithError(err).Fatal("Could not connect to RabbitMQ")
+	}
+
+	return conn
+}
+
+func connectionCleanup(shutdownHandler shutdown.ShutdownHandler, conn wabbit.Conn, log *logrus.Entry) {
```

Comment:
Can you add some comments for this func?

Reply:
Will do.

Reply:
Ended up deleting that function entirely.